modules/arm_plugin/src/arm_converter/arm_converter_quantize.cpp

```diff
@@ -9,10 +9,128 @@
+    NEQuantizationLayerQI &operator=(NEQuantizationLayerQI &&) = delete;
+    ~NEQuantizationLayerQI() = default;
+    void configure(const arm_compute::ITensor *input, arm_compute::ITensor *output, const arm_compute::QuantizationInfo &qi) {
+        ARM_COMPUTE_ERROR_ON_NULLPTR(input, output);
+        ARM_COMPUTE_ERROR_THROW_ON(NEQuantizationLayerQI::validate(input->info(), output->info(), qi));
+        _output = output;
+        _outputqi.allocator()->init(*(_output->info()));
+        _outputqi.info()->set_quantization_info(qi);
+        _quant = std::make_unique<arm_compute::NEQuantizationLayer>();
+        _quant->configure(input, &_outputqi);
+    }
+    static arm_compute::Status validate(const arm_compute::ITensorInfo *input, const arm_compute::ITensorInfo *output,
+                                        const arm_compute::QuantizationInfo &qi) {
+        ARM_COMPUTE_RETURN_ERROR_ON_NULLPTR(output);
+        //At the moment quantization info isn't checked actually, but just in case
+        return arm_compute::NEQuantizationLayer::validate(input, &arm_compute::TensorInfo(*output).set_quantization_info(qi));
+    }
+    void run() override {
+        ARM_COMPUTE_ERROR_ON_MSG(!_quant.get(), "Kernel didn't configured");
+        if (_outputqi.info()->padding() != _output->info()->padding()) _outputqi.info()->extend_padding(_output->info()->padding());
+        _outputqi.allocator()->import_memory(_output->buffer());
+        _quant->run();
+        _outputqi.allocator()->free();
+    }
+
+protected:
+    const arm_compute::ITensor *_output;
+    arm_compute::Tensor _outputqi;
+    std::unique_ptr<arm_compute::NEQuantizationLayer> _quant;
+};
+template <typename T, typename Q>
+void quantize(const T* input_data,
+              Q* output_data,
+              const ngraph::Shape& input_data_shape,
+              float scale,
+              std::int32_t offset) {
+    for (size_t i = 0; i < ov::shape_size(input_data_shape); ++i) {
+        float output = input_data[i] / scale + offset;
+        output -= std::remainder(output, 1.0f);
+        if (output < std::numeric_limits<Q>::lowest()) {
+            output_data[i] = std::numeric_limits<Q>::lowest();
+        } else if (output > std::numeric_limits<Q>::max()) {
+            output_data[i] = std::numeric_limits<Q>::max();
+        } else {
+            output_data[i] = static_cast<Q>(output);
+        }
+    }
+}
 template<> Converter::Conversion::Ptr Converter::Convert(const opset::ArmQuantize& node) {
-    return MakeConversion<arm_compute::NEQuantizationLayer>(node.input(0), node.output(0));
+    auto qInfoIt = node.get_rt_info().find("QuantizationInfo");
+    if (qInfoIt == node.get_rt_info().end()) {
+        IE_THROW() << "No quantization info available for ArmQuantize";
+    }
+    auto qInfo = safe_cast<ngraph::VariantWrapper<arm_compute::QuantizationInfo>>(qInfoIt->second)->get();
+    if (node.get_input_element_type(0).is_real()) {
```

Comment:
Move this check into transformation

Reply:
That's not a call to reference implementation. It's a local implementation that will require definition of separate operation to be controlled via transformations. I think it doesn't make sense to define two separate ArmQuantize operations used for non-intersecting data type sets just to make possible selection of a more proper implementation of the same operation in runtime.
By the way I'm going to investigate whether it would be possible to avoid use of scalar implementation at all.